modules/distr-tracing-tempo-object-storage-setup-ibm-storage.adoc

```diff
@@ -0,0 +1,131 @@
+
+.Procedure
+
+====
+
+. Fetch your credentials from the cluster deployed on {ibm-cloud-title}:
++
+[source,terminal]
+----
+export IBMCLOUD_API_KEY=$(oc get secrets ibmcloud-creds -n kube-system -o json | jq -r '.data.apiKey' | base64 -d)
+----
++
+. Log in to {ibm-cloud-title} by using {ibm-cloud-title} CLI:
++
+[source,terminal]
+----
+$ ibmcloud login -r us-east 
+----
++
+. Install the `cloud-object-storage` and `infrastructure-service` plugins:
++
+[source,terminal]
+----
+$ ibmcloud plugin install cloud-object-storage \
+  && \
+  ibmcloud plugin install infrastructure-service
+----
++
+. Create a resource group:
++
+[source,terminal]
+----
+$ ibmcloud resource group-create tempo-tracing \
+  && \
+  ibmcloud target -g tempo-tracing
+----
++
+. Create a service instance for the object store:
++
+[source,terminal]
+----
+$ ibmcloud resource service-instance-create -d premium-global-deployment \
+  <tempo_bucket> cloud-object-storage standard global \
+  && \
+  export IBM_SERVICE_INSTANCE_ID=$(ibmcloud resource service-instance <tempo_bucket> -o json | jq -r '.[0].crn')
+----
++
+. Create an object store bucket:
++
+[source,terminal]
+----
+$ ibmcloud cos bucket-create --bucket <tempo_bucket> --ibm-service-instance-id "$IBM_SERVICE_INSTANCE_ID"
+----
+====
+
+. Create a service key for connecting to the object store bucket:
++
+[source,terminal]
+----
+$ ibmcloud resource service-key-create <tempo_bucket> Writer --instance-name <tempo_bucket> --parameters '{"HMAC":true}'
+----
+
+====
+
+. Create a secret in the {product-title} cluster with the {ibm-cloud-title} bucket credentials:
++
+[source,terminal]
+----
+$ oc <project> <secret_name>
+----
++
+. Extract the access and secret keys to be used on secret creation:
++
+[source,terminal]
+----
+$ json_output=$(ibmcloud resource service-key <tempo_bucket> -o json) \
+  && \
+  export IBM_BUCKET_ACCESS_KEY=$(echo $json_output | jq -r '.[0].credentials.cos_hmac_keys.access_key_id') \
+  && \
+  export IBM_BUCKET_SECRET_KEY=$(echo $json_output | jq -r '.[0].credentials.cos_hmac_keys.secret_access_key') \
+  && \
+  export IBM_BUCKET_ENDPOINT="https://s3.<region>.cloud-object-storage.appdomain.cloud" # <1>
+----
+<1> For example, `us-east`.
+
+====
+
+. Create a secret on {ibm-cloud-title}:
++
+[source,terminal]
+----
+$ oc -n tracing-system create secret generic ibm-cos-secret \
+  --from-literal=bucket="<tempo_bucket>" \
+  --from-literal=endpoint="$IBM_BUCKET_ENDPOINT" \
+  --from-literal=access_key_id="$IBM_BUCKET_ACCESS_KEY" \
+  --from-literal=access_key_secret="$IBM_BUCKET_SECRET_KEY"
+----
++
+. Set the storage section in the TempoStack custom resource:
++
+[source,yaml]
+----
+apiVersion: tempo.grafana.com/v1alpha1
+kind: TempoStack
+# ...
+spec:
+# ...
+  storage:
+    secret:
+      name: <ibm-cos-secret> # <1>
```

Comment:
@IshwarKanse and @rubenvp8510, is this the OpenShift secret that is created in the step on line 82?

Reply:
yes